store-client/pkg/datastore/providers/postgresql/changestream.go

```diff
@@ -0,0 +1,1555 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresql
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/nvidia/nvsentinel/store-client/pkg/client"
+	"github.com/nvidia/nvsentinel/store-client/pkg/datastore"
+)
+
+const (
+	healthEventsTable = "health_events"
+	// PostgreSQL NOTIFY channel for instant change notifications
+	notifyChannel = "nvsentinel_changes"
+)
+
+// ChangeStreamMode defines the operating mode for the changestream watcher
+type ChangeStreamMode string
+
+const (
+	// ModePolling uses only polling (original implementation)
+	ModePolling ChangeStreamMode = "polling"
+	// ModeNotify uses only LISTEN/NOTIFY (experimental)
+	ModeNotify ChangeStreamMode = "notify"
+	// ModeHybrid uses LISTEN/NOTIFY with adaptive fallback polling (recommended)
+	ModeHybrid ChangeStreamMode = "hybrid"
+)
```

Comment:
Just for my understanding, is the `notify` option experimental for us or is that an exterimental/alpha feature in Postgres?

I'm also wondering if we can simplify and we can pick only one mode.